addons/product/models/res_config_settings.py

```diff
@@ -79,3 +74,8 @@ def _onchange_module_module_sale_product_matrix(self):
         If the user enables the Grid Configurator -> enable the product Configurator as well"""
         if self.module_sale_product_matrix and not self.module_sale_product_configurator:
             self.module_sale_product_configurator = True
+
+    def _toggle_group_multi_currency(self):
+        super()._toggle_group_multi_currency()
+        if self.group_multi_currency:
+            self.group_product_pricelist = True
```

Comment:
I don't think this is working because `_toggle_group_multi_currency` is called as an `api.model`: `self.env['res.config.settings']._toggle_group_multi_currency()`

Reply:
There is no reason for it to be in res.config.settings if it is an `api.model` - where will be a better place to move it? `res.currency`? I will adapt the override in product module too

Reply:
As in conversation above, this is not required and therefore removed. It is solved in the update to implied groups in product/security/product_security.xml